customer-service/src/main/scala/infrastructure/repositories/CustomerDetailsHttpRepository.scala

```diff
@@ -0,0 +1,70 @@
+package infrastructure.repositories
+
+import cats.data.OptionT
+import cats.effect.Async
+import cats.effect.kernel.{Fiber, Outcome}
+import cats.implicits._
+import domain.CustomerDetails
+import domain.repositories.CustomerDetailsRepository
+import infrastructure.config.Configs.CustomerDetailsServiceConfig
+import io.circe.generic.auto._
+import io.scalaland.chimney.dsl.TransformerOps
+import org.http4s.Method.GET
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.client.Client
+import org.http4s.{Request, Uri}
+
+final case class CustomerDetailsExternalApiOutput(data: String)
+final case class CustomerDetailsLegacyExternalApiOutput(data: String)
+
+class CustomerDetailsHttpRepository[F[_]: Async](config: CustomerDetailsServiceConfig, httpClient: Client[F])
+    extends CustomerDetailsRepository[F] {
+
+  private val root = Uri.unsafeFromString(
+    s"http://${config.customerDetailsConfig.host}:${config.customerDetailsConfig.port}/internal"
+  )
+
+  private val rootLegacyRepository = Uri.unsafeFromString(
+    s"http://${config.customerDetailsLegacyConfig.host}:${config.customerDetailsLegacyConfig.port}/internal/legacy"
+  )
+
+  private def handleRepositoryFibers(
+      res: Outcome[F, Throwable, Option[CustomerDetails]],
+      fiber: Fiber[F, Throwable, Option[CustomerDetails]],
+  ): F[Option[CustomerDetails]] = res match {
+    case Outcome.Succeeded(firstRepoDetails) =>
```

Comment:
now, the structure is good, however:

- u don't cancel other fiber if it is not needed:
```
case Some(_) => firstRepoDetails // other fiber is running even if u return here
```
cancel other fiber